app/src/test/kotlin/treesValidChecker/checkers.kt

```diff
@@ -0,0 +1,71 @@
+package treesValidChecker
+
+import trees.AVLTree
+import trees.BSTree
+import trees.RBTree
+import trees.nodes.AVLNode
+import trees.nodes.BSNode
+import trees.nodes.RBNode
+import kotlin.math.abs
+
+val BStree = BSTree<Int, Int>()
+
+fun isBstOk(node: BSNode<Int, Int>? = BStree.root): Boolean {
+    if (node == null) {
+        return true
+    }
+    if (node.left != null && node.left!!.key > node.key) {
```

Comment:
Во-первых, старайтесь никогда не использовать `!!`: если очень нужно, то лучше сделать `checkNotNull() { ... }` с понятным сообщением об ошибке
```suggestion
    if (node.left?.run { key > node.key } == true) {
```
Во-вторых, у вас же вроде бы невозможна ситуация, когда ключ в дереве повторяется, а значит, это тоже нужно отсекать. То есть, следует изменить `>` на `>=`.

Reply:
Аналогично с проверкой ниже